app/ydoc-server/src/languageServerSession.ts

```diff
@@ -646,7 +681,13 @@ class ModulePersistence extends ObservableV2<{ removed: () => void }> {
       this.syncedMeta = metadata
       this.syncedMetaJson = metadataJson
     }, 'file')
-    if (unsyncedIdMap) this.sendLsUpdate(contentsReceived, undefined, unsyncedIdMap, undefined)
+    if (unsyncedIdMap)
+      this.sendLsUpdate(
+        contentsReceived,
+        this.syncedCode ?? undefined,
+        unsyncedIdMap,
+        this.syncedMeta?.ide?.node,
```

Comment:
Providing the code argument to save the updated code snapshot after the successful metadata recovery